.github/actions/run-fixtures-test/action.yml

```diff
@@ -39,16 +39,32 @@ runs:
 
     - name: Generate a changelog
       working-directory: ${{ inputs.fixtures-dir }}
-      run: git cliff --config cliff.toml ${{ inputs.command }} > output.md
+      run: |
+        git cliff --config cliff.toml ${{ inputs.command }} > output.md || true
+        echo "$?" > output.rc
+        cat output.md
       shell: bash
 
     - name: Compare the output with the expected output
       working-directory: ${{ inputs.fixtures-dir }}
       env:
         IN_DATE_FORMAT: ${{ inputs.date-format }}
       run: |
-        cat output.md
-        current_date=$(date +"$IN_DATE_FORMAT")
-        sed -i "s/<<DATE>>/$current_date/g" expected.md
-        diff --strip-trailing-cr output.md expected.md
+        if [ -f expected.md ]; then
+          current_date=$(date +"$IN_DATE_FORMAT")
+          sed -i "s/<<DATE>>/$current_date/g" expected.md
+          if [ ! diff --strip-trailing-cr output.md expected.md ]; then
+            echo "The generated changelog does not match the expected changelog."
+            exit 1
+          fi
+        fi
+      shell: bash
+
+    - name: Compare the return code with the expected return code
```

Comment:
I'm thinking if we should support comparing the output log as well. It would really help to see what those error messages look like when unconventional commits are found.

Reply:
Good point, but somewhatt more complicated I assume. I wouldn't diff the entire log with some expected log, but only some messages.

Reply:
I have played around with this a bit and I couldn't find a clean solution. Matching on the entire output is too much. This would lead to random test failures whenever any log message is changed. The next step would be to have a file with expected log messages. But that too introduces problems. If a line contains any variable at all (e.g. the commit id), it will not match the expected output unless we do some magic string replacement first.

While I agree that this would be a nice thing to have, I'd opt for not tackling it in this PR.

Reply:
Good call. Created #1075.